workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/hooks/data/useGetVocabulariesAndCategories.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import useSWR from 'swr';
+
+import {ProductVocabulary} from '../../enums/ProductVocabulary';
+import HeadlessCommerceAdminCatalogImpl from '../../services/rest/HeadlessCommerceAdminCatalog';
+
+const useGetVocabulariesAndCategories = (vocabulariesName: string[]) => {
+	return useSWR({key: 'vocabularies/', vocabulariesName}, async () => {
+		const {
+			items,
+		} = await HeadlessCommerceAdminCatalogImpl.getMarketplaceVocabularies();
+
+		const _vocabularies = items.filter((vocabulary: ProductCategories) =>
+			vocabulariesName.includes(vocabulary.name as ProductVocabulary)
+		);
+
+		const vocabularies: any = {};
```

Comment:
Could you create the type for this

Reply:
We're gonna fix it in the next PR. Thanks!